### PR TITLE
Thread scheduling for AitkenNeville

### PR DESCRIPTION
**Pull request by saurabhkgp21**

### CODE REVIEW

File: src/perform_step/extrapolation_perform_step.jl [hunk 1]
```diff
@@ -142,19 +143,34 @@ function perform_step!(integrator,cache::AitkenNevilleConstantCache,repeat_step=
...
-      for j in 2:2^(i-1)
-        @muladd u_temp = @.. u_temp + dt_temp*k_temp
-        k_temp = f(u_temp, p, t+j*dt_temp)
+    # Balance workload of threads by computing T[1,1] with T[max_order,1] on
+    # same thread, T[2,1] with T[max_order-1,1] on same thread. Similarly fill
+    # first column of T matrix
+    Threads.@threads for i in 1:ceil(max_order/2)
+      # indices array hold row index of T matrix to be computed in this
+      # execution of thread
+      indices = [Int(i)]
```
Comment (ChrisRackauckas) [inline src/perform_step/extrapolation_perform_step.jl]: this will allocate more than you want. Re-structure to not have this array alloc.

Comment (saurabhkgp21) [reply src/perform_step/extrapolation_perform_step.jl]: Should we do the calculation of `T[i,1]` and `T[max_order+1-i,1]` by copy-pasting the same code twice? That will reduce the memory allocation though. Is there any other way around?

Comment (ChrisRackauckas) [reply src/perform_step/extrapolation_perform_step.jl]: You can make a tuple, but remember that its length is then type information, so always make it length 2 and check for equality.

**Comment by ChrisRackauckas** [comment]
that looks good to me. Is this complete then?

**Pull request merged**